unstructured/file_utils/filetype.py

```diff
@@ -107,24 +107,51 @@ def file_type(cls, ctx: _FileTypeDetectionContext) -> FileType:
     @property
     def _file_type(self) -> FileType:
```

Comment:
Why are these properties instead of lazyproperties?

Reply:
They could be I suppose. No compelling reason why not, but they're only called at most once so I went with keeping things simple. There's no code path from calling `_FileTypeDetector.file_type(ctx)` that would ever execute any of these more than once. Slightly simplifies the stack trace to use `@property`, but any reason to make them lazy would override that small benefit. Those reasons would be:
- computationally expensive and called more than once
- not idempotent, like it constructs a collaborator object you only want the one instance of.

It's a borderline call either way. If these were already lazyproperties and I had occasion to refactor, that wouldn't be something I'd be inclined to change unless one of the mentioned conditions applied.

It's also cheap and easy to change in the future if we need too. The interface is unchanged by it and it's a one line change.